flang/include/flang/Lower/CallInterface.h

```diff
@@ -217,12 +217,18 @@ class CallerInterface : public CallInterface<CallerInterface> {
     return procRef;
   };
   bool isMainProgram() const { return false; }
+  /// Returns true if this is a call to a procedure pointer of a dummy
+  /// procedure.
+  bool isIndirectCall() const;
 
   /// Helpers to place the lowered arguments at the right place once they
   /// have been lowered.
   void placeInput(const PassedEntity &passedEntity, mlir::Value arg);
   void placeAddressAndLengthInput(const PassedEntity &passedEntity,
                                   mlir::Value addr, mlir::Value len);
+  /// If this is a call to a procedure pointer/ dummy, returns the related symbol. Nullptr otherwise.
+  const Fortran::semantics::Symbol* getIndirectCallSymbol() const;
```

Comment:
Should this be a SymbolRef? We seem to use both arbitrarily.

Reply:
No, nullptr has a meaning here. I have renamed it to `getIfIndirectCallSymbol`.